src/components/ColorPicker.tsx

```diff
@@ -42,14 +49,74 @@ const ColorSwatch: FC<{
   const dispatch = useDispatch()
   const colors = useSelector(themeColors)
   const fontSize = useSelector(state => state.fontSize)
-  const selected =
-    (color && cursorStyle?.color === color) || (backgroundColor && cursorStyle?.backgroundColor === backgroundColor)
+  const currentThought = useSelector(state => !!state.cursor && getThoughtById(state, head(state.cursor))) as Thought
```

Comment:
It would be better to subscribe to the value of the thought rather than the whole thought, since that is all that is used. (You can use `head(cursor)` instead of `currentThought.id`).

Side note: `getThoughtById` does not return a stable reference, so as written this will cause the component to re-render on every change to the Redux state. This will be fixed when you subscribe to the value only, since strings are compared by value.